src/coreclr/vm/corelib.cpp

```diff
@@ -281,12 +281,6 @@ const USHORT c_nCoreLibFieldDescriptions = ARRAY_SIZE(c_rgCoreLibFieldDescriptio
 #define FCFuncElementSig(name,sig,impl) \
     FCFuncFlag_HasSignature + FCFuncElement(name, impl) (LPVOID)sig,
 
-#define FCIntrinsic(name,impl,intrinsicID) FCFuncFlags(intrinsicID, ECall::InvalidDynamicFCallId), \
-    (LPVOID)GetEEFuncEntryPoint(impl), (LPVOID)name,
-
-#define FCIntrinsicSig(name,sig,impl,intrinsicID) \
-    FCFuncFlag_HasSignature + FCIntrinsic(name,impl,intrinsicID) (LPVOID)sig,
-
 #define FCDynamic(name,intrinsicID,dynamicID) FCFuncFlags(intrinsicID, dynamicID), \
```

Comment:
You can also delete `intrinsicID` argument for `FCDynamic` and `FCFuncFlags` macros. It is not used anymore.

Reply:
Also, you can delete `ECall::GetIntrinsicID`. It will always return InvalidIntrinsic ID now.

Reply:
Looks like `FastAllocateString` is using `FCDynamic` in `ecallslist.h`.

``ByReference`1`` .ctor and get_Value intrinsic are still detected with `CORINFO_INTRINSIC_*` / `GetIntrinsicID` mechanism.

On a related note, there are also `CORINFO_INTRINSIC_Array_{Get,Address,Set}`, which are classified correctly on the runtime side, but as far as I can tell from dumps, they are always getting seen as `CORINFO_INTRINSIC_ByReference_{Ctor,Value}` in jit importer. I haven't find the root cause yet but somehow `HasSameTypeDefAs()` at https://github.com/dotnet/runtime/blob/562e7d11bffbab14a9b1516141475cf895aaa8a7/src/coreclr/vm/jitinterface.cpp#L8380-L8382 covers array cases too?

I was trying to convert ``ByReference`1`` intrinsics to NamedIntrinsic, but they are not striaght-forward as others in this PR; due to odd classification of array intrinsics.

Reply:
> So deleting the macro arg should not be a problem.

Ah, sorry I thought it was about deleting the whole definition. I'm testing the new patch locally.

>  They are not detected by `ECall::GetIntrinsicID`

I mixed up `ECall::GetIntrinsicID` and `CEEInfo::getIntrinsicID`, will delete.

> The array cases are covered by the `if (method->IsArray())` path a few lines above.

Yup, then the new if/else-block seem to overwrite `CORINFO_INTRINSIC_Array_*` classification as `CORINFO_INTRINSIC_ByReference_xx`. I couldn't find any instance of `CORINFO_INTRINSIC_Array_` like intrinsics with `DOTNET_JitDump=* corerun ..`, until I experimentally switched ByRefernce to NamedIntrinsic, which lead to code like `var myArray = new int[] { 1,2,2 }; myArray[2] = ; Console.WrieLine(myArray.Length);` to show `CORINFO_INTRINSIC_Array_` intrinsics showed up in the dumps (but crossgen2 started to fail with bad instruction errors).